### PR TITLE
Improving simulator test coverage

### PR DESCRIPTION
**Pull request by pcnfernando**
## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? yes
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

### CODE REVIEW

File: components/osgi-tests/src/test/java/org/wso2/carbon/analytics/test/osgi/SimulatorAPITestcase.java [hunk 1]
```diff
@@ -0,0 +1,301 @@
+package org.wso2.carbon.analytics.test.osgi;
```
Comment (mohanvive) [inline components/osgi-tests/src/test/java/org/wso2/carbon/analytics/test/osgi/SimulatorAPITestcase.java]: add license header.

File: components/osgi-tests/src/test/java/org/wso2/carbon/analytics/test/osgi/util/TestUtil.java [hunk 2]
```diff
@@ -16,124 +16,191 @@
...
-            HttpURLConnection urlConn = null;
-            try {
-                urlConn = TestUtil.generateRequest(baseURI, path, methodType, false);
-            } catch (IOException e) {
-                TestUtil.handleException("IOException occurred while running the HttpsSourceTestCaseForSSL", e);
+            URL url = baseURI.resolve(path).toURL();
+            boundary = "---------------------------" + currentTimeMillis();
+
+            connection = (HttpURLConnection) url.openConnection();
+          /*  connection.setConnectTimeout(CONNECT_TIMEOUT);
```
Comment (mohanvive) [inline components/osgi-tests/src/test/java/org/wso2/carbon/analytics/test/osgi/util/TestUtil.java]: remove commented codes.

File: components/osgi-tests/src/test/java/org/wso2/carbon/analytics/test/osgi/SimulatorAPITestcase.java [hunk 3]
```diff
@@ -0,0 +1,316 @@
...
+    private static final org.apache.log4j.Logger logger = org.apache.log4j.Logger.getLogger(SimulatorAPITestcase.class);
+
+    private static final int HTTP_PORT = 9390;
+    private static final String HOSTNAME = "localhost";
+    private URI baseURI = URI.create(String.format("http://%s:%d", HOSTNAME, HTTP_PORT));
+
+    private final String DEFAULT_USER_NAME = "admin";
+    private final String DEFAULT_PASSWORD = "admin";
+
+    private static final String validSingleEventConfig = "{\n" +
```
Comment (tishan89) [inline components/osgi-tests/src/test/java/org/wso2/carbon/analytics/test/osgi/SimulatorAPITestcase.java]: Constant names should be uppercase. Applies for below variables as well

File: components/org.wso2.carbon.event.simulator.core/src/main/java/org/wso2/carbon/event/simulator/core/internal/generator/SingleEventGenerator.java [hunk 4]
```diff
@@ -185,21 +191,23 @@ private static SingleEventSimulationDTO createSingleEventDTO(String singleEventC
...
-//            create SingleEventSimulationDTO
             SingleEventSimulationDTO singleEventSimulationDTO = new SingleEventSimulationDTO();
             singleEventSimulationDTO.setStreamName(singleEventConfig.getString(EventSimulatorConstants.STREAM_NAME));
             singleEventSimulationDTO.setSiddhiAppName(singleEventConfig
                     .getString(EventSimulatorConstants.EXECUTION_PLAN_NAME));
             singleEventSimulationDTO.setTimestamp(timestamp);
             singleEventSimulationDTO.setAttributeValues(dataValues.toArray());
             return singleEventSimulationDTO;
-        } catch (JSONException e) {
+        }catch (JsonProcessingException e) {
```
Comment (ramindu90) [inline components/org.wso2.carbon.event.simulator.core/src/main/java/org/wso2/carbon/event/simulator/core/internal/generator/SingleEventGenerator.java]: Can you fix the formatting issue?

File: components/osgi-tests/src/test/java/org/wso2/carbon/analytics/test/osgi/SimulatorAPITestcase.java [hunk 5]
```diff
@@ -0,0 +1,316 @@
...
+    protected BundleContext bundleContext;
+
+    @Configuration
+    public Option[] createConfiguration() {
+        return new Option[]{
+                carbonDistribution(Paths.get("target", "wso2das-" +
+                        System.getProperty("carbon.analytic.version")), "editor"),
+                copySiddhiFileOption(),
+                copyCSVFileOption(),
+                //CarbonDistributionOption.debug(5005)
```
Comment (ramindu90) [inline components/osgi-tests/src/test/java/org/wso2/carbon/analytics/test/osgi/SimulatorAPITestcase.java]: Shall we remove the unnecessary comments?

File: components/osgi-tests/src/test/java/org/wso2/carbon/analytics/test/osgi/SimulatorAPITestcase.java (hunk 3 above)
Comment (pcnfernando) [reply components/osgi-tests/src/test/java/org/wso2/carbon/analytics/test/osgi/SimulatorAPITestcase.java]: Fixed with commit [#9e041b8abf0bee874137cdcbafd38e6bde8a4930](https://github.com/wso2/carbon-analytics/pull/1259/commits/9e041b8abf0bee874137cdcbafd38e6bde8a4930)

File: components/org.wso2.carbon.event.simulator.core/src/main/java/org/wso2/carbon/event/simulator/core/internal/generator/SingleEventGenerator.java (hunk 4 above)
Comment (pcnfernando) [reply components/org.wso2.carbon.event.simulator.core/src/main/java/org/wso2/carbon/event/simulator/core/internal/generator/SingleEventGenerator.java]: Fixed with commit [#4eb71832d44025049ade5f9e651eadea3fc556d5](https://github.com/wso2/carbon-analytics/pull/1259/commits/4eb71832d44025049ade5f9e651eadea3fc556d5)

File: components/osgi-tests/src/test/java/org/wso2/carbon/analytics/test/osgi/SimulatorAPITestcase.java (hunk 5 above)
Comment (pcnfernando) [reply components/osgi-tests/src/test/java/org/wso2/carbon/analytics/test/osgi/SimulatorAPITestcase.java]: Fixed with commit [#f9917365bd8f8a1c65a09dace8e464fae8933ec3](https://github.com/wso2/carbon-analytics/pull/1259/commits/f9917365bd8f8a1c65a09dace8e464fae8933ec3)

**Pull request merged**